.circleci/config.yml

```diff
@@ -1,23 +1,34 @@
 version: 2.1
 
 orbs:
-  python: circleci/python@0.2.1
+  python: circleci/python@2.1.1
+
 jobs:
   install_and_update_dependencies:
     docker:
-      - image: circleci/python:3.8.6-node
+      - image: cimg/python:3.8.6-node
     working_directory: ~/repo
     steps:
       - checkout
       - restore_cache:
           keys:
-            - v2-dependencies-{{ checksum "Pipfile.lock" }}-{{ checksum "thumbs/Pipfile.lock" }}
-            - v2-dependencies-default
-      - run: &install-pipenv sudo pip install -U urllib3 cryptography==3.3.2 pipenv pip --quiet --no-input
-      - run: sudo pip uninstall -y poetry
-      - run: pipenv install --dev
-      - run: pipenv check # before save_cache so an insecure cache is never saved
+            - v3-dependencies-{{ checksum "Pipfile.lock" }}-{{ checksum "thumbs/Pipfile.lock" }}
+            - v3-dependencies-default
+      # Ensure pip and pipenv are installed
+      - run:
+          name: Install pipenv
+          command: |
+            python -m pip install --upgrade pip urllib3 cryptography==3.3.2 --quiet --no-input
+            pip install pipenv
+      - run: pip uninstall -y poetry
+      # Ensure that pipenv will create the virtual environment in the project directory
+      - run:
+          name: Set up virtual environment
+          command: |
+            export PIPENV_VENV_IN_PROJECT=1
+            pipenv install --dev
```

Comment:
Use `pipenv sync --dev` to install in CI. `sync` installs from `Pipfile.lock`. `install` defers to `Pipfile` and re-locks.